dali/pipeline/data/view_as_higher_ndim.h

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef  DALI_PIPELINE_DATA_VIEW_AS_HIGHER_NDIM_H_
+#define  DALI_PIPELINE_DATA_VIEW_AS_HIGHER_NDIM_H_
+
+#include <utility>
+#include <vector>
+#include "dali/pipeline/data/views.h"
+
+namespace dali {
+
+/**
+ * @brief Inserts `insert_ndim` new dimensions with extent one to a given position in a shape
+ */
+static TensorListShape<> insert_unit_dims(const TensorListShape<> &sh, int insert_ndim,
+                                          int position) {
```

Comment:
```suggestion
template <int out_ndim = DynamicDimensions, int in_ndim>
inline TensorListShape<out_ndim> insert_unit_dims(const TensorListShape<in_ndim> &sh, int position, int count) {
```
...and move to `tensor_shape.h`

Reply:
As discussed offline, renamed to `expand_dims` and moved to dali/core